Wrappers/Python/test/test_DataContainer.py

```diff
@@ -88,6 +74,88 @@ def test_ndim(self):
         self.assertEqual(x_np.ndim, x_cil.ndim)
         self.assertEqual(3, x_cil.ndim)
 
+    def test_DataContainer_equal(self):
+        array = np.linspace(-1, 1, 32, dtype=np.float32).reshape(4, 8)
+        data = DataContainer(array)
+        data1 = data.copy()
+
+        # Check two identical DataContainers are equal
+        self.assertTrue((data == data1).all())
+        # Check it works for comparing DataContainer with numpy array
+        self.assertTrue((data == data1.as_array()).all())
```

Comment:
should we not use `np.testing.assert_array_equal` or similar?

Reply:
https://github.com/TomographicImaging/CIL/blob/a181c35dc853b4985283f86af3900f869166147e/Wrappers/Python/cil/framework/data_container.py#L898-L902

So `.all` of a boolean array should be enough.

Reply:
np.testing.assert_array_equal does not work because data and data1 aren't arrays they're data containers. When I try to use it I get error:

```
Traceback (most recent call last):
  File "/home/lhe97136/miniconda3/envs/cil_dev2/lib/python3.12/site-packages/numpy/testing/_private/utils.py", line 742, in assert_array_compare
    val = comparison(x, y)
          ^^^^^^^^^^^^^^^^
ValueError: The truth value of an array with more than one element is ambiguous. Use a.any() or a.all()


Arrays are not equal
 x: array(<cil.framework.data_container.DataContainer object at 0x7f20f2f86e70>,
      dtype=object)
 y: array(<cil.framework.data_container.DataContainer object at 0x7f20f2f86fc0>,
      dtype=object)
Traceback (most recent call last):
  File "/home/lhe97136/miniconda3/envs/cil_dev2/lib/python3.12/site-packages/numpy/testing/_private/utils.py", line 742, in assert_array_compare
    val = comparison(x, y)
          ^^^^^^^^^^^^^^^^
ValueError: The truth value of an array with more than one element is ambiguous. Use a.any() or a.all()
During handling of the above exception, another exception occurred:
Traceback (most recent call last):
  File "/home/lhe97136/miniconda3/envs/cil_dev2/lib/python3.12/unittest/case.py", line 58, in testPartExecutor
    yield
  File "/home/lhe97136/miniconda3/envs/cil_dev2/lib/python3.12/unittest/case.py", line 634, in run
    self._callTestMethod(testMethod)
  File "/home/lhe97136/miniconda3/envs/cil_dev2/lib/python3.12/unittest/case.py", line 589, in _callTestMethod
    if method() is not None:
       ^^^^^^^^
  File "/home/lhe97136/CIL/Wrappers/Python/test/test_DataContainer.py", line 84, in test_DataContainer_equal
    np.testing.assert_array_equal(data, data1)
  File "/home/lhe97136/miniconda3/envs/cil_dev2/lib/python3.12/site-packages/numpy/testing/_private/utils.py", line 920, in assert_array_equal
    assert_array_compare(operator.__eq__, x, y, err_msg=err_msg,
  File "/home/lhe97136/miniconda3/envs/cil_dev2/lib/python3.12/contextlib.py", line 81, in inner
    return func(*args, **kwds)
           ^^^^^^^^^^^^^^^^^^^
  File "/home/lhe97136/miniconda3/envs/cil_dev2/lib/python3.12/site-packages/numpy/testing/_private/utils.py", line 805, in assert_array_compare
    raise ValueError(msg)
ValueError: 
error during assertion:
Traceback (most recent call last):
  File "/home/lhe97136/miniconda3/envs/cil_dev2/lib/python3.12/site-packages/numpy/testing/_private/utils.py", line 742, in assert_array_compare
    val = comparison(x, y)
          ^^^^^^^^^^^^^^^^
ValueError: The truth value of an array with more than one element is ambiguous. Use a.any() or a.all()
Arrays are not equal
 x: array(<cil.framework.data_container.DataContainer object at 0x7f20f2f86e70>,
      dtype=object)
 y: array(<cil.framework.data_container.DataContainer object at 0x7f20f2f86fc0>,
````

Reply:
Sorry I didn't understand the second comment about .all?